pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
+		Elem:          &schema.Schema{Type: schema.TypeString},
+		Optional:      true,
+		Description:   "Specifies the list of roles that the client can set as the primary role.",
+		ConflictsWith: []string{"external_oauth_blocked_roles_list"},
 	},
-	"audience_urls": {
+	"external_oauth_audience_list": {
 		Type:        schema.TypeSet,
 		Elem:        &schema.Schema{Type: schema.TypeString},
 		Optional:    true,
 		Description: "Specifies additional values that can be used for the access token's audience validation on top of using the Customer's Snowflake Account URL ",
 	},
-	"any_role_mode": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Default:     string(snowflake.Disable),
-		Description: "Specifies whether the OAuth client or user can use a role that is not defined in the OAuth access token.",
-		ValidateFunc: validation.StringInSlice([]string{
-			string(snowflake.Disable),
-			string(snowflake.Enable),
-			string(snowflake.EnableForPrivilege),
-		}, true),
+	"external_oauth_any_role_mode": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  fmt.Sprintf("Specifies whether the OAuth client or user can use a role that is not defined in the OAuth access token. Valid options are: %v", sdk.AsStringList(sdk.AllExternalOauthSecurityIntegrationAnyRoleModes)),
+		ValidateFunc: validation.StringInSlice(sdk.AsStringList(sdk.AllExternalOauthSecurityIntegrationAnyRoleModes), true),
 		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
 			normalize := func(s string) string {
 				return strings.ToUpper(strings.ReplaceAll(s, "-", ""))
 			}
 			return normalize(old) == normalize(new)
 		},
 	},
-	"scope_delimiter": {
+	"external_oauth_scope_delimiter": {
 		Type:        schema.TypeString,
 		Optional:    true,
 		Description: "Specifies the scope delimiter in the authorization token.",
 	},
+	"external_oauth_scope_mapping_attribute": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies the access token claim to map the access token to an account role.",
+	},
 	"comment": {
 		Type:        schema.TypeString,
 		Optional:    true,
 		Description: "Specifies a comment for the OAuth integration.",
 	},
-	"created_on": {
-		Type:        schema.TypeString,
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW SECURITY INTEGRATIONS` for the given security integration.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowSecurityIntegrationSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
 		Computed:    true,
-		Description: "Date and time when the External OAUTH integration was created.",
+		Description: "Outputs the result of `DESCRIBE SECURITY INTEGRATIONS` for the given security integration.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeExternalOauthSecurityIntegrationSchema,
+		},
+	},
+	ParametersAttributeName: {
```

Comment:
Shouldn't this be `RelatedParametersAttributeName`?

Reply:
Changed.